### DISCUSSION TITLE
KAMP on SV06

### DISCUSSION THREAD

**Discussion by Heiko910** [bassamanator/Sovol-SV06-firmware#96] [General]
Hi there,
I used the Bassamanator Methode to setup Klipper on my SV06.
I also looked at all the included configs as well as macros, KAMP and such are already included in the Bassamanator master kit.
I would love to use adaptive bed mesh and adaptive purging but I just can't figure out how to actually make it work. 
I am using prusaslicer and I made sure label object is enabled.
Any help would be appreciated.
Thanks

**Comment by bassamanator** [comment]
Please see https://github.com/bassamanator/Sovol-SV06-firmware#how-do-i-enable-kamp-klipper-adaptive-meshing--purging

**Reply by Heiko910** [reply]
Thank you sir. I appreciate your help. I think I forgot to enable variable_kamp. 
I have a print running right now but I will try that out ASAP.
Thanks again,
Heiko

**Reply by Heiko910** [reply]
That was it. I got the variable mesh and the purge running. I even mentioned to change the start point of the purge slightly so the x axis doesn't crash into the endstop that hard.
Thanks again for al the hard work you put into this project.
Next thing I am going to try is to install the hotend from a SV06 Plus onto my regular SV06 for more extrusion speed. 

I will also add a synchronization belt to the top of the z axis lead screws to keep the two motors synchronized when level. they seem to go out of alignment pretty quickly and the gantry align ramming into the top doesn't seem to align anything on my printer.

Thanks again,
Heiko

**Comment by bassamanator** [comment]
You're welcome.

Yeah a plus hotend is an easy way to get more plastic flowing. A cht nozzle/clone also helps.

I'm curious, how do you know the motors are getting out of alignment, poor first layers?

**Reply by Heiko910** [reply]
If I do a gantry level with G34 and then do a bed mesh the right side of the bed is always showing really low, which means the z axis is really high on the right.
If I then home all axis - Disable the steppers - move the print head over to the right side manually - lower the z axis until the inductive sensor triggers - raise it up just a bit until the led turns back on - move over to the left do the same - do those steps a few time left and right and then home all axis again and repeat the bed mesh, I get a much better result. I am using the inductive probe to level the z axis to the bed surface. That works great but I really have to do that every time the print was off or the steppers were disabled.

If I don't and I just check with a bed mesh the right side bed is low again/the z axis is high on the right. 

The left side of the z axis is always low. I am assuming that is because the left side carries the weight of the x axis stepper. 
I will level the gantry with the probe and then connect the left and right lead screw with a timing belt.
A guy from France already did all the leg work. He designed a belt tensioner that mounts in the middle of the gantry. I will let you know how that works out. If that turns out to be to troublesome, I will go the route to replace the control board and separate the z axis steppers. But that is a project for future Heiko.
Let me know if you have any input on this. 
Greetings,
Heiko

**Comment by bassamanator** [comment]
I was going to do the sync mod originally, it took me a few tries to get the correct belt size, but then I realized that it was not needed. But maybe it's needed on your unit? If you press down *firmly* on the left, and then the right, x axis gantry plastic molded pieces, is there any movement? Test this while the extruder is on the far right/left, and the center.
![Screenshot_20231011_150144](https://github.com/bassamanator/Sovol-SV06-firmware/assets/61985779/88006650-e5d1-4d06-a407-7d8b15a7770a)

**Reply by Heiko910** [reply]
When I get home tonight I will try that. What am I looking for? What if there is movement? Was something loose in your case?

**Comment by Heiko910** [comment]
Do you remember the belt size that you came up with? I just ordered a few different sizes last night.
But I am sure I will have try a few before it all fits. I just finished printing all the parts for the belt tensioner system. 
This the link to the belt tensioner 
https://cults3d.com/en/3d-model/tool/synchronisation-de-l-axe-z-pour-la-sovol-sv06-plus

**Comment by bassamanator** [comment]
Somewhere too far to look for, but based on my order history, the 690mm belt loop should fit perfectly with the idlers that I have, they were a 'standard' size. Since you're using a belt tensioner, you can go much larger.

No, for the gantry, things were not loose in my case. It's just very unusual if the z-motors are going out of sync, print to print. Something else is at play here. I would suggest you run through the initial steps on the everything sv06 repo, and then after that, check to see if your top beam is square on its own. The user @BeauSlim found that his top beam was twisted slightly, which lead to all sorts of issues.

**Reply by Heiko910** [reply]
Interesting. Some YouTube reviewer called Maker's Muse said that even the Prusa printers that look very similar, have the same issue. When the printer is turned off and the steppers are not powered anymore, then the z axis steppers go out of alignment. 
The left side has the added weight of the x axis stepper.

**Reply by BeauSlim** [reply]
@Heiko910 I wonder if you are seeing something I saw on mine, where my right Z nut wasn't actually riding on the bottom of the Z screw thread.  The geometry of the X rods and tension from the belt was pulling up on my right Z carriage box, especially when the X carriage was on the left side.  Keeping the X belt tension relatively low and installing (4mm lead) POM anti-backlash Z nuts seemed to solve it.  https://www.aliexpress.com/item/1005002212236572.html

The nuts also stop the Z motors from going out of sync because they grip a bit more.  I only need to level my X gantry every couple weeks.

**Reply by Heiko910** [reply]
That is some good information. Let me check if that is the case. I actually ended up adding a belt and tensioner between the two z axis lead screws. The belt now keeps the lead screws in perfect alignment and it seems to be working out just fine.


![IMG20231016141258.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/5d3544db-d56b-487c-92c9-84b07e53feb4)

![IMG20231016141303.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/7ec92713-6eef-4693-934c-435e10c895c9)

![IMG20231016141310.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/09da62fb-7dc5-4763-b6bb-5ac1baf99add)

![IMG20231016141307.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/39987fbe-62cb-438f-9ca2-824291e066f9)

![IMG20231016141506.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/b08269c5-e49c-4201-a93c-31cea2c2649a)

**Reply by bassamanator** [reply]
@BeauSlim Hey slim. What do you think is happening here as far as desynchronization is concerned? Do you think it's a matter of motor microsteps that are out of sync adding up, over time, to full steps? The source of the misalignment could be attributed to uneven pressures applied to the lead screws, which are obviously connected to the z motors. I guess this might explain any out of sync issues that you personally experienced, or is there something else at play here?

**Comment by MacBandito** [comment]
I’ve been lurking on this thread.  I too have this issue.  But it only happens after I’ve turn my printer off and back on.  The I have to re-align my z.  Turns out I’ve had those issue since I got the machine several months ago but didn’t realize what was going on until recently.  Since I noticed I’ve been doing searches and there’s a lot of people with this issue and several other chasing a problem they don’t know the cause of but it’s pretty likely this. 

Also I’ve seen multiple solutions for this from the timing belt which locks both z motors to adding control for the second motor and the simplest solution being finding level then using a sharpie mark the timing of the z rod clamp on top of the drive motor housing so that before a print you can at a glance see if it’s lost timing and simply move it back if it has. It’s usually no more than a 1/4 turn off.

**Reply by Heiko910** [reply]
I have tried the gantry level that the machine is designed for and that did NOT bring the left and right z axis lead screws into alignment.
I then used the probe for left and right alignment and that works well, but as others have mentioned that procedure would have to be redone each time the machine is turned off or the steppers were off for a period of time.  
I also used the sharpie Methode but ended up never relying on it.

I contemplated if I should convert my printer to a different control board that has two independent z axis stepper drivers but ended up adding the timing belt between the two lead screws.
That seems to have done the trick. The z axis lead screws move relatively slow as the print progresses and there is not much strain on that new timing belt.
For the standard SV06 the correct size belt is 690mm. I currently have 696mm and that works too.
The belt tensioner can be purchased on Cults3d.
I modified that design slightly and added heat thread inserts in a couple of places.


See pictures

![IMG20231016141310.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/891ce1b4-bd72-4aef-8f76-393c78b94b31)

![IMG20231016141303.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/50da3def-c169-4e8b-9423-a0f3368e4624)

![IMG20231016141307.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/25f1054f-7f62-4e7f-881b-d98ec518b7d7)

![IMG20231016141506.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/fed99248-8f26-4b34-9e93-e748b5c990da)

![IMG20231016141258.jpg](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/0512279b-f5c3-40ef-a389-1e04da13cb37)

**Reply by MacBandito** [reply]
That’s very nice add on. I was going to do the same mod but realized I could just mark the motor position once it was aligned.  There is a file someone made you can print to give you a fancy pointer to mark the position as well.

**Reply by Heiko910** [reply]
thanks for sharing_

**Reply by brogrene** [reply]
This is the video I'd seen about z-markers. [Z-markers to keep Z-motors aligned on Sovol SV06, SV06Plus + How to Z-align using inductive probe](https://youtu.be/VI7icch4VEM?si=GDSdPcyQwbQEezDk)

**Comment by Heiko910** [comment]
First of it only happens when the machine is off or the steppers are off
for a while, like over night. On my sv06 it is always the left side that is
low.
I think that happens because the x axis stepper motor is hanging on the
left side and that adds uneven weight to that side.
As long as I go from print to print after leveling it with the inductive
probe, it is fine. Only when the machine is off over night it goes out of
alignment.
Mechanical gantry leveling doesn't correct that at all.
I usually home the machine, then I turn the steppers off and then I use the
triggering of the probe to level the z axis on the left and right side of
the bed. Once I am done with that I home the machine again.

Now I don't have to do that anymore since I added a belt between the two z
axis lead screws.
Cheers,

**Comment by BeauSlim** [comment]
Wow that is quite the mod.  Cool that it lets you get the screws parallel.  My concern with this is that linking the 2 Z motors you are over-constraining the mechanism, and I wonder if there may be side-effects.   I'm very much a "whatever works" guy so if prints look good and you don't burn out a stepper motor after a couple months, great.

**Reply by Heiko910** [reply]
On the SV06 the two z axis steppers are wired in parallel to one driver on the control board. They are supposed to do the exact same at the same time from the factory. What my mod does is just keep them from sagging independently when the machine is off.

Another thing that I just did is to print the z axis stepper motor mounts, because the factory brackets were not square to the vertical parts of the printer. This caused a misalignment in the z axis lead screw. With my printed version it is perfectly square now.

![IMG20231018225419](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/5ae63d0d-f4b7-4327-af2f-780a8138da96)


![IMG20231018225738](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/d58bd87c-7952-4ad7-972c-ea8a53ff70ca)
